app/views/dashboards/_project.html.erb

```diff
@@ -0,0 +1,8 @@
+<tr class="<%= 'warning' if deploy_group_versions.values.uniq.count > 1 %>">
+  <td><%= link_to project.name, project_path(project) %></td>
+  <% @environment.deploy_groups.each do |deploy_group| %>
+    <td>
+      <%= link_to deploy_group_versions[deploy_group.id].reference, project_deploy_path(project, deploy_group_versions[deploy_group.id]) if deploy_group_versions[deploy_group.id] %>
```

Comment:
Aye, I just can't bring myself to do assignment within a conditional check. Years of c/c++ bugs have rejected that pattern.
I'll have a look at creating a helper method for this.